calendar/calendar-tool/tool/src/webapp/vm/calendar/chef_calendar_view.vm

```diff
@@ -22,3 +22,38 @@
 <script src="/library/webjars/fullcalendar/5.8.0/main.min.js"></script>
 <script src="/library/webjars/fullcalendar/5.8.0/locales-all.min.js"></script>
 <script src="/sakai-calendar-tool/js/sakai-calendar.js"></script>
+
+<script>
+  // This logic is associated to set the default subview, by day, month, week or list.
+  const defaultSubview = '$defaultSubview';
+  switch (defaultSubview) {
+    case 'day':
+      calendar.changeView('timeGridDay');
+      break;
+    case 'month':
+      calendar.changeView('dayGridMonth');
+      break;
+    case 'list':
+      calendar.changeView('listWeek');
+      break;
+    case 'week':
+    default:
+      calendar.changeView('timeGridWeek');
+      break;
+  }
+
+  // When the user changes the view, reflect the change in a param to set the default view.
+  setDefaultView = () => {
```

Comment:
this makes me a bit twitchy. You're setting this on the window object, and it's pretty generic sounding. It could clash. I'd call it setCalendarDefaultView.

Reply:
This whole script block should really do in a file. Why not just add it to sakai-calendar.js?

Reply:
Because that logic is more related to the calendar views and what the backend stores for the default view, I didn't want to mess with the full calendar script :(

Reply:
I'd still put it in a js file though as it's a substantial chunk of code. Caching and linting are made possible then. You should definitely change the name of that function, too. It's just too generic to have in the window scope.

Reply:
I'd personally still just put it in the same file as the full calendar instantiation. Also, looking at sakai-calendar.js, you have a lot of global stuff at the top of that file that may well be defined elsewhere on the page by other tool's js. I'd add this js to that file and namespace it under something. Maybe create a calendar object and put everything under that, or sakaiCalendar, perhaps. It just makes our js more robust when we namespace it.

Reply:
You could also look at using optional chaining in your portal object traversals. You get a short circuited undefined if a parent node doesn't exist. So, window.top?.portal?.user?.timezone won't blow up, it'll just return undefined earlier.